src/Nox.Solution/Models/Domain/Entities/Entity.cs

```diff
@@ -69,5 +70,103 @@ internal bool ApplyDefaults()
             if (string.IsNullOrWhiteSpace(PluralName)) PluralName = Name.Pluralize();
             return true;
         }
+
+        public IEnumerable<KeyValuePair<EntityMemberType, NoxSimpleTypeDefinition>> GetAllMembers()
+        {
+            foreach (var key in Keys!)
+            {
+                yield return new (EntityMemberType.Key, key);
+            }
+            
+            if (Attributes is not null ) 
+            { 
+                foreach (var attribute in Attributes)
+                {
+                    yield return new (EntityMemberType.Attribute, attribute);
+                }
+            }
+
+            if (OwnedRelationships is not null)
+            {
+                foreach (var relationship in OwnedRelationships)
+                {
+                    if (relationship.Related.Entity?.Keys is null)
+                    {
+                        if (relationship.Relationship == EntityRelationshipType.OneOrMany || relationship.Relationship == EntityRelationshipType.ZeroOrMany)
+                        {
+                            var foreignKeyName = $"{relationship.Related.Entity!.PluralName}";
```

Comment:
The code blocks inside "if" and "else" are practically identical - maybe it makes sense to have only foreignKeyName and Type logic inside if/else block to highlight the actual difference.